src/Autocomplete.tsx

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable @typescript-eslint/indent */
+import * as React from 'react';
+import { useState } from 'react';
+import { Person } from './types/Person';
+import debounce from 'lodash.debounce';
+
+interface AutocompleteProps {
+  people: Person[];
+  delay?: number;
+  onSelected: (person: Person | null) => void;
+}
+
+export const Autocomplete: React.FC<AutocompleteProps> = ({
+  people,
+  delay = 300,
+  onSelected,
+}) => {
+  const [inputValue, setInputValue] = useState('');
+  const [suggestions, setSuggestions] = useState<Person[]>([]);
+  const [isDropDownActive, setIsDropDownActive] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const filterSuggestions = debounce((value: string) => {
+    const results = value
+      ? people.filter(person =>
+          person.name.toLowerCase().includes(value.toLowerCase()),
+        )
+      : people;
+
+    setSuggestions(results);
+    setIsDropDownActive(true);
+  }, delay);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setInputValue(value);
+    filterSuggestions(value);
+
+    if (selectedPerson) {
+      setSelectedPerson(null);
+      onSelected(null);
+    }
+  };
+
+  const handleSelectPerson = (person: Person) => {
+    setSelectedPerson(person);
+    setInputValue(person.name);
+    setSuggestions([]);
+    setIsDropDownActive(false);
+    onSelected(person);
+  };
+
+  const handleInputFocus = () => {
+    if (!inputValue) {
+      setSuggestions(people);
+      setIsDropDownActive(true);
+    }
```

Comment:
The `handleInputFocus` function sets suggestions to all people when the input is focused and empty. This is a good approach to show all options initially, but ensure that this behavior aligns with the desired user experience. If not, consider adjusting the logic to better fit the requirements.